File: executable/greeter/src/main/swift/hello.swift
#if FRENCH
func greeting() -> String {
  return "Bonjour, Monde!"
}
#endif


public class Greeter {
  public init() {}
  public func sayHello() {
    #if FRENCH
    print(greeting())
    #else
    print("Hello, World!")
    #endif
  }
}

File: cpp/simple-library/src/main/cpp/hello.cpp
#include "hello.h"

#include <string>
#include <iostream>

#if FRENCH
std::string greeting() {
  return "Bonjour, Monde!";
}
#endif

void Greeter::sayHello() {
#if FRENCH
    std::cout << greeting() << std::endl;
#else
    std::cout << "Hello, World!" << std::endl;
#endif
}
